drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -253,12 +261,26 @@ static int flash_npcx_nor_read(const struct device *dev, off_t addr,
 	return 0;
 }
 
-static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
-				  size_t size)
+static void disable_multiple_erase_sizes(void)
+{
+#undef ERASE_BLOCK_64K
+#undef ERASE_BLOCK_4K
+}
+
+static void enable_multiple_erase_sizes_within_range(const struct flash_npcx_nor_config *config)
+{
+	if (config->min_erase_size <= BLOCK_64K_SIZE && BLOCK_64K_SIZE <= config->max_erase_size) {
+#define ERASE_BLOCK_64K
+	}
+	if (config->min_erase_size <= BLOCK_4K_SIZE && BLOCK_4K_SIZE <= config->max_erase_size) {
+#define ERASE_BLOCK_4K
+	}
```

Comment:
I am quite sure this is not how C preprocessor works.

Reply:
Actually I tried testing with 0x10000 as max and min erase size, then trying to erase 0x1000 size, the erase failed. But when 0x1000 is min and 0x10000 is max, the 0x1000 erase happened successfully

Reply:
Still, I am quite sure this is not how C preprocessor works.

Reply:
I removed macros at runtime in new revision